### DISCUSSION TITLE
vercel dev requires authentication, preventing usage from CI

### DISCUSSION THREAD

**Discussion by ThisIsMissEm** [vercel/vercel#8618] [Ideas]
Hi!

We're currently trying to setup our vercel/vite app for end-to-end testing with playwright on GitHub Actions for CI, and we've noticed that `vercel dev` always requires authentication.

Would it be possible to skip this if `$CI` is set, or a `--skip-linking-envs` flag is passed, or something like that?

I did find `$__VERCEL_SKIP_DEV_CMD` in the code that appears to do what I want, but it looks very undocumented.

**Comment by amyegan** [comment]
Thanks for reaching out @ThisIsMissEm! Can you tell me a little more about your workflow? I don't want to give bad advice based on faulty assumptions. 

There is a guide on [how to use GitHub Actions with Vercel](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel) which includes a section with steps to add the required values from Vercel as secrets in GitHub. The [token option](https://vercel.com/docs/cli/global-options#token) might be what you need.

**Reply by ThisIsMissEm** [reply]
Right, so, both those don't quite cover what we're doing. We're only using `vercel dev` as we've a serverless function that's part of our app. We want to run end-to-end tests on our app as part of CI/CD, and the app is started via Playwright.

We don't want to be pulling environment variables or anything else from Vercel, we're good to keep that all in GitHub.

You can see what I mean here: https://github.com/inrupt/client-identifier-helper/blob/main/playwright.config.ts#L67-L73

For now, we've worked around by adding a "CI" user account to our Vercel project, and then using it's token as a secret in GitHub Actions, but honestly, this is complicated and poses security risks (an additional account to secure that has production access); It'd be much better to just say to vercel "hey, just use what ever is in the current environment, don't try to phone home to vercel and pull environment variables"

hence a `--skip-env-sync` or something along those lines which would just disable these lines of code: https://github.com/vercel/vercel/blob/main/packages/cli/src/commands/dev/dev.ts#L32-L78

**Reply by amyegan** [reply]
Thanks for this clarification. I see now how this is a bit different from what is currently available. I'll ask the CLI team to take a look! Adding to the requested features list.
